### ISSUE TITLE
Want to support the Embedded.Empty(Nullable) annotation

### ISSUE THREAD

**Issue by DreamStar92** [infobip/infobip-spring-data-querydsl#102]
I forked the project and found that the configuration of the Embedded annotation is upstream, and since the upstream does not handle data-jdbc, this needs to be handled by infobip-querydsl-apt. I'm not sure at the moment how the appropriate modification is, so wanted to discuss it. 

I can provide PR if you want to add such a feature.

**Comment by lpandzic** [comment]
So basically Q class support for embedded classes? Sure, feel free to create a PR.

**Comment by DreamStar92** [comment]
@Embedded(onEmpty = OnEmpty.USE_EMPTY) = @Embedded.Empty

@Embedded(onEmpty = OnEmpty.USE_NULL) = @Embedded.Nullable

What I mean by this is that the correct Q class is generated when the entity uses a convenience annotation.

**Comment by DreamStar92** [comment]
Due to time zone issues I may not be able to reply to you in time, I plan to implement the following logic.
(Since Embedded is nestable, prefix must be continuously appended, just like what data-jdbc itself implements.)
Can you give me some guidance?

```java
public class Inventory {
    @Embedded.Empty(prefix = "prefix_")
    private EarlyWarningRange earlyWarningRange;
}
public record EarlyWarningRange(
        int averageSalesDays,
        int earlyWarningMaxLimitDays,
        int earlyWarningMinLimitDays
) {}

public class QInventory extends com.querydsl.sql.RelationalPathBase<Inventory> {
    public final NumberPath<Integer> averageSalesDays = createNumber("averageSalesDays", Integer.class);
    public final NumberPath<Integer> earlyWarningMaxLimitDays = createNumber("earlyWarningMaxLimitDays", Integer.class);
    public final NumberPath<Integer> earlyWarningMinLimitDays = createNumber("earlyWarningMinLimitDays", Integer.class);

    public void addMetadata() {
        addMetadata(averageSalesDays, ColumnMetadata.named("prefix_average_sales_days").withIndex(0));
        addMetadata(earlyWarningMaxLimitDays, ColumnMetadata.named("prefix_early_warning_max_limit_days").withIndex(1));
        addMetadata(earlyWarningMinLimitDays, ColumnMetadata.named("prefix_early_warning_min_limit_days").withIndex(2));
    }
}
```

**Comment by lpandzic** [comment]
> Can you give me some guidance?

Basic case is here - [Embedded](https://github.com/infobip/infobip-spring-data-querydsl?tab=readme-ov-file#JDBCEmbeddedSupport). 
I'd first start with a test like this one - https://github.com/infobip/infobip-spring-data-querydsl/blob/master/infobip-spring-data-jdbc-annotation-processor/src/test/java/com/infobip/spring/data/jdbc/annotation/processor/EntityWithEmbedded.java
Expected result for this case is here 
https://github.com/infobip/infobip-spring-data-querydsl/blob/master/infobip-spring-data-jdbc-annotation-processor/src/test/resources/expected/QEntityWithEmbedded.java

So basically write the test and what you expect Querydsl to generate here and then go on from there. You can debug the annotation processor and figure out what is going on.

**Comment by DreamStar92** [comment]
Sorry I haven't given feedback for a long time. Due to work reasons, I don't have time to contribute to this feature.

Here's my solution to my current work dilemma and my apologies for not keeping my end of the bargain.

Currently, dynamic query is implemented through the Query mechanism in spring data jdbc's own JdbcAggregateOperations.

In terms of query only, JdbcClient is used to integrate RelationalConverter to realize dynamic query and integration with the converter part of data-jdbc. It is a lightweight mechanism using native SQL.

**Comment by lpandzic** [comment]
> Sorry I haven't given feedback for a long time. Due to work reasons, I don't have time to contribute to this feature.
> 
> Here's my solution to my current work dilemma and my apologies for not keeping my end of the bargain.
> 
> Currently, dynamic query is implemented through the Query mechanism in spring data jdbc's own JdbcAggregateOperations.
> 
> In terms of query only, JdbcClient is used to integrate RelationalConverter to realize dynamic query and integration with the converter part of data-jdbc. It is a lightweight mechanism using native SQL.

Understandable, it's similar for everyone working on open source projects on the side.
Thank you for sharing your solution.
